storage/http_client.go

```diff
@@ -178,10 +180,45 @@ func (c *httpStorageClient) ListBuckets(ctx context.Context, project string, opt
 // Bucket methods.
 
 func (c *httpStorageClient) DeleteBucket(ctx context.Context, bucket string, conds *BucketConditions, opts ...storageOption) error {
-	return errMethodNotSupported
+	s := callSettings(c.settings, opts...)
+	req := c.raw.Buckets.Delete(bucket)
+	setClientHeader(req.Header())
+	if err := applyBucketConds("httpStorageClient.DeleteBucket", conds, req); err != nil {
+		return err
+	}
+	if s.userProject != "" {
+		req.UserProject(s.userProject)
+	}
+
+	return run(ctx, func() error { return req.Context(ctx).Do() }, s.retry, s.idempotent)
```

Comment:
How do the settings know the default idempotency value for this delete operation?

Reply:
Communicating the idempotency will the responsibility of the transport-agnostic client layer (i.e. the layer above these transport-specific implementations that will call them). it will communicated to this transport-specific layer via the `storageOption.idempotent(bool)`. This is basically the same as it is today, where the transport-agnostic client layer supplies `true` or `false` to the `run` call. Swap out the `run` call for the transport-client interface call with the option `idempotent(true)` and we are good to do.